apps/trading/e2e/actions/utils.py

```diff
@@ -35,4 +35,29 @@ def next_epoch(vega: VegaServiceNull):
                 "Epoch not started after forwarding the duration of two epochs."
             )
     vega.wait_fn(1)
-    vega.wait_for_total_catchup()
\ No newline at end of file
+    vega.wait_for_total_catchup()
+
+def wait_for_graphql_response(page, query_name, timeout=5000):
+    response_data = {}
+
+    def handle_response(route, request):
+        if "graphql" in request.url:
+            response = request.response()
+            if response is not None:
+                json_response = response.json()
+                if json_response and "data" in json_response:
```

Comment:
same for here really. This method is potentially a black hole for errors, unless we want it to be totally passive for some reason